Command fireball giving error

/fireball is giving error in console and it wont summon a fireball.
<img width="1150" alt="Screen Shot 2020-02-18 at 1 15 38 AM" src="https://user-images.githubusercontent.com/47665139/74709660-bbea1280-51ed-11ea-818c-c9e13d948e5c.png">
this is the error...
https://pastebin.com/J85x9dJ5

I can reproduce this bug on the following:
```
[17:33:27 INFO]: CONSOLE issued server command: /ess version
[17:33:27 INFO]: Server version: 1.8.8-R0.1-SNAPSHOT git-PaperSpigot-"4c7641d" (MC: 1.8.8)
[17:33:27 INFO]: EssentialsX version: 2.17.1.61
[17:33:27 INFO]: Vault is not installed. Chat and permissions may not work.
```

config: https://gist.github.com/triagonal/a3fa35bbcded1e508cc244f5913f0740 (default)
server startup: https://gist.github.com/triagonal/389f120fedc976b7ba4b8e0aef786da7

When running `/fireball` for the first time (from either the console or ingame), this error appears:
```
[17:32:49 INFO]: HelloHarry issued server command: /fireball
[17:32:49 ERROR]: Command fireball failed:
java.lang.NoClassDefFoundError: org/bukkit/entity/DragonFireball
        at com.earth2me.essentials.commands.Commandfireball.<clinit>(Commandfireball.java:22) ~[EssentialsX-2.17.1.61.jar:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_241]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_241]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_241]
        at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_241]
        at java.lang.Class.newInstance(Unknown Source) ~[?:1.8.0_241]
        at com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:529) [EssentialsX-2.17.1.61.jar:?]
        at com.earth2me.essentials.Essentials.onCommand(Essentials.java:470) [EssentialsX-2.17.1.61.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) [patched.jar:git-PaperSpigot-"4c7641d"]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) [patched.jar:git-PaperSpigot-"4c7641d"]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchCommand(CraftServer.java:666) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnection.handleCommand(PlayerConnection.java:1189) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1001) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:45) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:1) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-PaperSpigot-"4c7641d"]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_241]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_241]
        at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-PaperSpigot-"4c7641d"]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_241]
Caused by: java.lang.ClassNotFoundException: org.bukkit.entity.DragonFireball
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:92) ~[patched.jar:git-PaperSpigot-"4c7641d"]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:87) ~[patched.jar:git-PaperSpigot-"4c7641d"]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_241]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_241]
        ... 24 more
```

On any subsequent run, this (slightly different) error appears instead:
```
[17:33:11 INFO]: HelloHarry issued server command: /fireball
[17:33:11 ERROR]: Command fireball failed:
java.lang.NoClassDefFoundError: Could not initialize class com.earth2me.essentials.commands.Commandfireball
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_241]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_241]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_241]
        at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_241]
        at java.lang.Class.newInstance(Unknown Source) ~[?:1.8.0_241]
        at com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:529) [EssentialsX-2.17.1.61.jar:?]
        at com.earth2me.essentials.Essentials.onCommand(Essentials.java:470) [EssentialsX-2.17.1.61.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) [patched.jar:git-PaperSpigot-"4c7641d"]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) [patched.jar:git-PaperSpigot-"4c7641d"]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchCommand(CraftServer.java:666) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnection.handleCommand(PlayerConnection.java:1189) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1001) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:45) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:1) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-PaperSpigot-"4c7641d"]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_241]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_241]
        at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-PaperSpigot-"4c7641d"]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-PaperSpigot-"4c7641d"]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_241]
```

--- a/Essentials/src/com/earth2me/essentials/commands/Commandfireball.java
+++ b/Essentials/src/com/earth2me/essentials/commands/Commandfireball.java
@@ -2,6 +2,7 @@ package com.earth2me.essentials.commands;
 
 import com.earth2me.essentials.User;
 import com.earth2me.essentials.utils.FloatUtil;
+import com.earth2me.essentials.utils.VersionUtil;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Lists;
 import org.bukkit.Server;
@@ -16,19 +17,31 @@ import java.util.stream.Collectors;
 import static com.earth2me.essentials.I18n.tl;
 
 public class Commandfireball extends EssentialsCommand {
-    private static final Map<String, Class<? extends Entity>> types = ImmutableMap.<String, Class<? extends Entity>>builder()
-        .put("fireball", Fireball.class)
-        .put("small", SmallFireball.class)
-        .put("large", LargeFireball.class)
-        .put("dragon", DragonFireball.class)
-        .put("arrow", Arrow.class)
-        .put("skull", WitherSkull.class)
-        .put("egg", Egg.class)
-        .put("snowball", Snowball.class)
-        .put("expbottle", ThrownExpBottle.class)
-        .put("splashpotion", SplashPotion.class)
-        .put("lingeringpotion", LingeringPotion.class)
-        .build();
+    private static final Map<String, Class<? extends Projectile>> types;
+
+    static {
+        ImmutableMap.Builder<String, Class<? extends Projectile>> builder = ImmutableMap.<String, Class<? extends Projectile>>builder()
+                .put("fireball", Fireball.class)
+                .put("small", SmallFireball.class)
+                .put("large", LargeFireball.class)
+                .put("arrow", Arrow.class)
+                .put("skull", WitherSkull.class)
+                .put("egg", Egg.class)
+                .put("snowball", Snowball.class)
+                .put("expbottle", ThrownExpBottle.class)
+                .put("splashpotion", SplashPotion.class)
+                .put("lingeringpotion", LingeringPotion.class);
+
+        if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_9_R01)) {
+            builder.put("dragon", DragonFireball.class);
+        }
+
+        if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_13_0_R01)) {
+            builder.put("trident", Trident.class);
+        }
+
+        types = builder.build();
+    }
 
     public Commandfireball() {
         super("fireball");
@@ -48,7 +61,8 @@ public class Commandfireball extends EssentialsCommand {
             try {
                 speed = FloatUtil.parseDouble(args[1]);
                 speed = Double.max(0, Double.min(speed, ess.getSettings().getMaxProjectileSpeed()));
-            } catch (Exception ignored) {}
+            } catch (Exception ignored) {
+            }
         }
 
         if (args.length > 2 && args[2].equalsIgnoreCase("ride") && user.isAuthorized("essentials.fireball.ride")) {
@@ -60,7 +74,7 @@ public class Commandfireball extends EssentialsCommand {
         }
 
         final Vector direction = user.getBase().getEyeLocation().getDirection().multiply(speed);
-        Projectile projectile = (Projectile) user.getWorld().spawn(user.getBase().getEyeLocation().add(direction.getX(), direction.getY(), direction.getZ()), types.get(type));
+        Projectile projectile = user.getWorld().spawn(user.getBase().getEyeLocation().add(direction.getX(), direction.getY(), direction.getZ()), types.get(type));
         projectile.setShooter(user.getBase());
         projectile.setVelocity(direction);
 
@@ -73,8 +87,8 @@ public class Commandfireball extends EssentialsCommand {
     protected List<String> getTabCompleteOptions(final Server server, final User user, final String commandLabel, final String[] args) {
         if (args.length == 1) {
             return types.keySet().stream()
-                .filter(type -> user.isAuthorized("essentials.fireball." + type))
-                .collect(Collectors.toList());
+                    .filter(type -> user.isAuthorized("essentials.fireball." + type))
+                    .collect(Collectors.toList());
         } else if (args.length == 2) {
             return Lists.newArrayList("1", "2", "3", "4", "5");
         } else {

